### PR TITLE
#279 show course with grades more than 100

### PR DESCRIPTION
**Pull request by pushyamig**
Relates to #279

### CODE REVIEW

File: dashboard/views.py [hunk 1]
```diff
@@ -216,8 +216,13 @@ def grade_distribution(request, course_id=0):
         return HttpResponse(json.dumps({}), content_type='application/json')
     number_of_students = df.shape[0]
     df = df[df['current_grade'].notnull()]
-    df['current_grade_mod'] = df['current_grade'].apply(lambda x: x.replace("100","99.99"))
-    average_grade = df['current_grade'].astype(float).mean().round(2)
+    df['current_grade'] = df['current_grade'].astype(float)
+    if df[df['current_grade'] > 100.0].shape[0] > 0:
+        df['more_than_100_percent'] = True
```
Comment (zqian) [inline dashboard/views.py]: is the new ''more_than_100_percent' column with bool value used later in tool?

File: dashboard/templates/grades.html [hunk 2]
```diff
@@ -14,12 +14,12 @@
...
         {% endblock %}
-        {% block js %}
-            <script src="{% static 'd3-v4/build/d3.min.js' %}"></script>
-            <script src="/static/js/grade_distribution.js"></script>
-        {% endblock %}
     {% else %}
         Grade Distribution view is hidden for this course.
     {% endif %}
 {% endblock %}
+{% block js %}
```
Comment (zqian) [inline dashboard/templates/grades.html]: not sure why this change is needed for current issue: if grade distribution is hidden, no js is needed, right?

File: dashboard/static/js/grade_distribution.js [hunk 3]
```diff
@@ -37,9 +38,10 @@ var makeGraph = function () {
         height = height - margin.top - margin.bottom;
         g = svg.append("g")
             .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+        var domain_values = (data_more_than_100_percent) ? [0, 110] : [0, 100]
```
Comment (zqian) [inline dashboard/static/js/grade_distribution.js]: Is 110 big enough to accommodate extra points?

File: dashboard/templates/grades.html (hunk 2 above)
Comment (pushyamig) [reply dashboard/templates/grades.html]: This is bug that resulted from JS error is the console  as this added the <script> tags twice in the html as resulted i guess from the  toggle view changes . So this is not related to current issue but I felt might as well fix it since I am here

File: dashboard/views.py (hunk 1 above)
Comment (pushyamig) [reply dashboard/views.py]: Yes

File: dashboard/static/js/grade_distribution.js (hunk 3 above)
Comment (jonespm) [reply dashboard/static/js/grade_distribution.js]: Yeah, maybe it would be nice to see what the visualization looks like in this case. 

Extra credit in grading has somewhat confused me. Either there's still a hard limit of 100% and anything over it doesn't get you anything extra, or grades over 100% give you something different. It looks like in Canvas, the grading schema can't go over 100% but students can still score over 100%. https://community.canvaslms.com/thread/2127

I wonder if our visualization needs to account for this? Maybe have a little "piece" on the end that's "up-to" 10% in width with a tooltip that says how far over 100% the students total grade is? Is this is what this code is attempting to do? Maybe an idea for phase 2.

Comment (pushyamig) [reply dashboard/static/js/grade_distribution.js]: Should be good or have to do the math based on the percent appropriate. I could also tell D3 that fix the domain limit based on values in the list. The problem with that is if the class is in 90-100 the distribution range it looks odd.  I will lie if I say i did not think is 110 enough? But being practical will the % grade ever grow higher than  110%, I felt unlikely so I made the hardcoded number.

Comment (pushyamig) [reply dashboard/static/js/grade_distribution.js]: screen shot 2018-11-02 at 9 30 30 am
screen shot 2018-11-02 at 9 30 46 am

Comment (jonespm) [reply dashboard/static/js/grade_distribution.js]: Yeah this looks good. This 110% only shows if there are grades over 100% right? I think it's nice to show. If there are any grades over 110% it could probably change the label to "110%+" and just show them on the "110%+ line". We might need to add this logic to the files page too. 

I think we *could* have just even left it at 100% and put all the grades >= 100% on a "100%+" line but it's probably nice to have this extra bit of visualization.

Comment (jonespm) [reply dashboard/static/js/grade_distribution.js]: Just wait till we have the gamification class that goes up to 1000% or something someday. ;)

Comment (pushyamig) [reply dashboard/static/js/grade_distribution.js]: OK, I have made the upper limit adaptive based on the data. so if we have course with grade distribution like 150% the view will serve it than abandoning scores as before
@jonespm I like your idea of making it something like 110%+, I think code needs to change the logic around `MyGrade` line if a student has a score 115% (say) that means you i have put the orange line on 110%+ marker as that is the last marker on the line. So I feel the docs needs to be updated with this feature. We will keep it simple for now for everybody. We can think about this idea for next phase.

Comment (DanielDugan) [reply dashboard/static/js/grade_distribution.js]: I agree with the above comments. I think a 110%+ column is the best visualization option. We push the distribution of the graph to far to the left in scenarios where a handful of students score in 1.4, 1.5, and 10.1 percentages.
